extensions/vscode/src/views/homeView.ts

```diff
@@ -90,6 +92,14 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
   private webviewView?: WebviewView;
   private extensionUri: Uri;
   private webviewConduit: WebviewConduit;
+  private initCompleteResolver: (() => void) | undefined = undefined;
+
+  // Promise that methods can wait on, which will be resolved when
+  // the initial queries finish. Then they can use our internal collections of
+  // contentRecords, credentials and configs within this class
+  private initComplete: Promise<void> = new Promise((resolve) => {
+    this.initCompleteResolver = resolve;
+  });
```

Comment:
This functionality allows for methods to wait on this promise to be resolved. We have to save the the related resolve method to this particular promise, so that when we resolve it, those with a reference to the original promise will be signalled on its completion.